frontend/javascripts/oxalis/controller/combinations/tool_controls.ts

```diff
@@ -785,6 +792,139 @@ export class QuickSelectTool {
+  static volumeLayer: VolumeLayer | null | undefined = null;
+  static getPlaneMouseControls(
+    _planeId: OrthoView,
+    planeView: PlaneView,
+    showNodeContextMenuAt: ShowContextMenuFunction,
+  ): any {
+    const SceneController = getSceneController();
+    const { quickSelectAreaGeometry } = SceneController;
+    return {
+      leftMouseDown: (pos: Point2, plane: OrthoView, _event: MouseEvent) => {
+        if (plane === OrthoViews.TDView) {
+          return;
+        }
+
+        const state = Store.getState();
+        this.volumeTracing = getActiveSegmentationTracing(state);
+        const maybeResolutionAndZoomStep = getRenderableResolutionForSegmentationTracing(
+          state,
+          this.volumeTracing,
+        );
+        if (!maybeResolutionAndZoomStep || !this.volumeTracing) {
+          return;
+        }
+        this.initialPlane = plane;
+        const position = V3.floor(calculateGlobalPos(state, pos));
+        const thirdDimValue = position[Dimensions.thirdDimensionForPlane(plane)];
+        this.volumeLayer = new VolumeLayer(
+          this.volumeTracing.tracingId,
+          plane,
+          thirdDimValue,
+          maybeResolutionAndZoomStep.resolution,
+        );
+        Store.dispatch(maybePrefetchEmbeddingAction(position));
+      },
+      leftDownMove: (
+        _delta: Point2,
+        pos: Point2,
+        id: string | null | undefined,
+        evt: MouseEvent,
+      ) => {
+        console.log("leftDownMove");
+        if (evt.altKey) {
+          MoveHandlers.moveWhenAltIsPressed(_delta, pos, id, evt);
+          return;
+        }
+        if (id == null || !this.volumeTracing || !this.volumeLayer) {
+          return;
+        }
+        if (!this.isDrawingBounds) {
+          this.initialPlane = id as OrthoView;
+          this.isDrawingBounds = true;
+          quickSelectAreaGeometry.reset();
+          quickSelectAreaGeometry.show();
+          quickSelectAreaGeometry.setViewport(id as OrthoView);
+          Store.dispatch(resetContourAction());
+        }
+        if (id !== this.initialPlane) {
+          return;
+        }
+        const state = Store.getState();
+        const position = V3.floor(calculateGlobalPos(state, pos, this.initialPlane));
+        quickSelectAreaGeometry.addEdgePoint(position);
+        Store.dispatch(addToContour(position));
+        this.volumeLayer.addContour(position);
+      },
+      leftMouseUp: () => {
+        const cleanUp = () => {
+          Store.dispatch(resetContourAction());
+          quickSelectAreaGeometry.resetAndHide();
+        };
+        const _state = Store.getState();
+        if (!this.isDrawingBounds || !this.volumeLayer) {
+          return;
+        }
+        if (
+          quickSelectAreaGeometry.vertexBuffer.getLength() <= 2
+          // || state.uiInformation.quickSelectState === "inactive"
+          // Dont know why this check is here ^
+        ) {
+          cleanUp();
+          return;
```

Comment:
@philippotto Sorry, but I do not clearly understand the purpose of this check. I copied this from the rectangle variation of the tool. Is this check necessary here? What does the state save / what for is `quickSelectState`?

Reply:
Yes, this is not super clear. I also had too look it up.
Quoting from the store:
```
readonly quickSelectState:
    | "inactive"
    | "drawing" // the user is currently drawing a bounding box
    | "active"; // the quick select saga is currently running (calculating as well as preview mode)
```

So, in general, it should not be `inactive` when the mouse button is lifted, since mouseDown should have made the state not inactive.
However, I think, the var might be `inactive` in the following cases:
- the tool is canceled with `escape` while the mouse button is pressed. then, it is lifted.
- the mouse is pressed in another tool and while it's being pressed, the tool is changed via a keyboard shortcut. then, the mouse button is lifted.
- a bug/error occurs somewhere.

Feel free to add a comment to the code to make it easier to understand :)